TMS.ConsoleApp/Entities/Training.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using TMS.ConsoleApp.Interfaces;
+
+namespace TMS.ConsoleApp.Entities
+{
+    internal class Training : EntityBase, IDeepClonable<Training>
+    {
+        public List<EntityBase> trainingTypes = new List<EntityBase>();
+        private Training _training;
+
+        public void Add(EntityBase training)
+        {
+            trainingTypes.Add(training);
+        }
+
+        public bool IsPractical()
+        {
+            if (trainingTypes.Count == 0)
+            {
+                return false;
+            }
+            if (trainingTypes.All(x => x is PracticalLesson))
+            {
+                return true;
+            }
+            return false;            
+        }
+        //https://stackoverflow.com/questions/129389/how-do-you-do-a-deep-copy-of-an-object-in-net
+        public Training Clone()
+        {
+            var trainingClone = new Training();
+
+            if (_training != null)
+            {
+                trainingClone._training = _training.Clone(); //reference type .Clone() that does the same
+            }
+           
+            return trainingClone;
```

Comment:
The clone should contain same Description property and list of EntityBase. 
Current implementation clones only private property of the Training type  and do not see it is used actually .

To make clone, do the ctor and create new instance:
I can`t recall if Igor asked to-do the deep clone, but assume that it is not required. 
`
public Training Clone()
{
    return new Training(this.Description, this.trainingType);  
}
`

Reply:
About Cloning: I created "Copy constructor" where I re-initialize the props:
  public Training(string description, List<EntityBase> trainingTypes)
        {
            Description = description;

            _trainingTypes = trainingTypes;
        }
Implement Clone() like so: (with new I am creating a new memory address in the Heap)
         public Training Clone()
        {
            return new Training(this.Description, this._trainingTypes); // returning new obj of type (Training)
        }